### DISCUSSION TITLE
Optimising svg paths

### DISCUSSION THREAD

**Discussion by noahbald** [parcel-bundler/lightningcss#816] [Show and tell]
Hello lightningcss folks, I'm writing a little svgo clone in Rust. I've been using lightningcss for various optimisations there already and I was really hoping that it also had something for optimising paths.
Obviously it doesn't, when parsing something like a `clip-path` with `path(...)` I think I recall it being parsed as `Unparsed` or `Custom` or something like that.

But anyway, I'm not here to complain. I ended up writing my own path optimiser. I always like giving back, so I'm here to ask if it'd be worth building this out more so you can use it.
I'm trying to structure it out so it could eventually be a crate and if I can get it properly tested and reasonably stable I can come help y'all integrate it.

You can check out the path optimiser [here](https://github.com/noahbald/oxvg/tree/main/crates/oxvg_path), and if you want an example of it being used you can take a look [here](https://github.com/noahbald/oxvg/blob/main/crates/oxvg_optimiser/src/jobs/convert_path_data.rs).

**Comment by devongovett** [comment]
Ooh nice work! I've been wanting to build an SVG optimizer based on lightning css for parsing style attributes, markup5ever, etc. for a while but haven't had time to work on it. Really cool to see.

Yes, it would be awesome to integrate a path optimizer for clip-path etc. As long as it's available as an independent crate without too many additional dependencies I'd love to use it in Lightning CSS!

**Comment by noahbald** [comment]
Love to hear it. Right now there's only two essential dependencies for paths: bitflags and ryu.
Also, as part of the general project, conversions between shapes (e.g `rect(...)` -> `path(...)`) may be possible
